framework/scenario/src/facade/scenario_world_runner.rs

```diff
@@ -0,0 +1,49 @@
+use crate::ScenarioWorld;
+use multiversx_chain_vm::scenario::{handler::StepRunner, model::*};
+
+impl ScenarioWorld {
+    pub fn for_each_runner_mut<F: FnMut(&mut dyn StepRunner)>(&mut self, mut f: F) {
+        f(&mut self.vm_runner);
+        if let Some(trace) = &mut self.trace {
+            f(trace);
+        }
+    }
+}
+
+impl StepRunner for ScenarioWorld {
+    fn run_external_steps(&mut self, step: &ExternalStepsStep) {
+        self.for_each_runner_mut(|runner| runner.run_external_steps(step));
+    }
+
+    fn run_set_state_step(&mut self, step: &SetStateStep) {
```

Comment:
Also, all these steps like `SetStateStep` are implemented in `vm`.

Reply:
This PR was not about moving, just yet.